### PR TITLE
Add specifications about pipeline output

### PR DESCRIPTION
**Pull request by chuan-wang**

### CODE REVIEW

File: conf/docker.config [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * Defines basic usage limits and docker image id.
+ * Imported under the default 'docker' Nextflow
+ * profile in nextflow.config
+ */
+
+docker {
+  enabled = true
+}
+process {
+  container = 'scilifelab/ngi-smRNAseq'
```
Comment (ewels) [inline conf/docker.config]: Docker containers can only be lower case. So `scilifelab/ngi-smrnaseq`

File: tests/docker_test.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+docker -v >/dev/null 2>&1 || { echo >&2 "I require docker, but it's not installed. Visit https://www.docker.com/products/overview#/install_the_platform  ."; exit 1; }
+nextflow -v >/dev/null 2>&1 || { echo >&2 "I require nextflow, but it's not installed. If you hava Java, run 'curl -fsSL get.nextflow.io | bash'. If not, install Java."; exit 1; }
+
+data_dir=${data_path}/ngi-smrna_test_set
+if [ -d $data_dir ]
+then
+    echo "Found existing test set, using $data_dir"
+else
+    echo "Downloading test set..."
+    curl https://export.uppmax.uu.se/b2013064/test-data/ngi-smrnaseq_test_set.tar.bz2 > ${data_path}/ngi-smrnaseq_test_set.tar.bz2
```
Comment (ewels) [inline tests/docker_test.sh:29]: Have you generated this data? How did you make it?

File: tests/docker_test.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    echo "Unpacking test set..."
+    tar xvjf ${data_path}/ngi-smrnaseq_test_set.tar.bz2 -C ${data_path}
+    echo "Done"
+fi
+
+nf_cmd="nextflow run $script_path -resume -profile docker --genome GRCh37 --reads \"${data_dir}/*.R1.fastq.gz\""
```
Comment (ewels) [inline tests/docker_test.sh]: The docker profile doesn't support `--genome`, so you need to specify the path to each index here instead. These need to be included in the test dataset that you download above.

**Review by ewels — COMMENTED** [review]
Looks really good! The Dockerfile seems to be pretty similar to the ones we've already made, so you may be in luck 😉

File: tests/docker_test.sh (hunk 2 above)
Comment (chuan-wang) [reply tests/docker_test.sh:29]: I picked four samples from the GA project G.A1610_16_01. The files are pretty small.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,101 @@
+FROM openjdk:8
+
+LABEL author="Phil Ewels" \
```
Comment (ewels) [inline Dockerfile:3]: You made this one! ;)

File: tests/docker_test.sh (hunk 2 above)
Comment (ewels) [reply tests/docker_test.sh:29]: Ok great - I just looked and it's 4.4GB. Whilst small for a project, this is pretty big for a test dataset (we'll be pulling it a lot so it should be as small and fast as possible).

Can you use `seqtk` to downsample the FastQ files? eg. to subsample down to 100 thousand reads:
```bash
module load seqtk
seqtk sample file.fastq.gz 100000 > file_sub100k.fastq
gzip file_sub100k.fastq
```
This should be sufficient to run the pipeline.

File: tests/docker_test.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    echo "Found existing test set, using $data_dir"
+else
+    echo "Downloading test set..."
+    curl https://export.uppmax.uu.se/b2013064/test-data/ngi-smrnaseq_test_set.tar.bz2 > ${data_path}/ngi-smrnaseq_test_set.tar.bz2
+    echo "Unpacking test set..."
+    tar xvjf ${data_path}/ngi-smrnaseq_test_set.tar.bz2 -C ${data_path}
+    echo "Done"
+fi
+
+nf_cmd="nextflow run $script_path -resume -profile docker --genome GRCh37 --mature ${data_dir}/mature_GRCh37 --hairpin ${data_dir}/hairpin_GRCh37 --bt2index ${data_dir}/genome --gtf ${data_dir}/genes.gtf --reads \"${data_dir}/*.R1.fastq.gz\""
```
Comment (ewels) [inline tests/docker_test.sh]: The travis builds are now running as I enabled Travis - see here:
https://travis-ci.org/SciLifeLab/NGI-smRNAseq/builds/217147411

Cool! 🎉 

However, this shows that I was right about it failing with `--genome` 😉  You've added the paths for the references, but you need to remove `--genome GRCh37`

File: tests/docker_test.sh (hunk 2 above)
Comment (chuan-wang) [reply tests/docker_test.sh:29]: Actually the test data is pretty small, but the hg19 reference files are big. Do you think that we can keep only part of the hg19 ref?

File: Dockerfile (hunk 4 above)
Comment (chuan-wang) [reply Dockerfile:3]: No no, I just mimicked your scripts XD. I think you should keep all credits.

File: tests/docker_test.sh (hunk 5 above)
Comment (chuan-wang) [reply tests/docker_test.sh]: I will fix this one.

**Comment by ewels** [comment]
Ok, the Travis build failure is now just because the ngi-smrnaseq docker image isn't published. I've just created this public docker repo, which will automatically build from GitHub: https://hub.docker.com/r/scilifelab/ngi-smrnaseq/

It's currently empty, but should trigger as soon as this PR is merged.

**Comment by chuan-wang** [comment]
Good! Thanks! I was googling how to solve this...

**Pull request merged**